graphics/libfive/Portfile

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cmake 1.1
+
+# libfive does not have releases; the master branch is recommended.
+# See https://libfive.com/download/
+github.setup        libfive libfive f471c12ea6766ed07cc99597423018ded8b238e7
+version             20210427
+
+name                libfive
+categories          graphics math cad
+platforms           darwin
+license             MPL-2
+maintainers         nomaintainer
+
+homepage            https://libfive.com/
+
+description         Library for 3D solid modeling
+
+long_description    libfive is a software library and set of tools for solid modeling, \
+                    especially suited for parametric and procedural design. It is \
+                    infrastructure for generative design, mass customization, and \
+                    domain-specific CAD tools.
+
+checksums           rmd160  9e09a15d2503704166b20ed1f00bc6356ecf772a \
+                    sha256  1a508bc19611f48ffc407b18611af2bb035a11c1be53632945e630d0c51e5a73 \
+                    size    877924
+
+depends_lib-append  port:eigen3 \
+                    port:libpng \
+                    port:boost
+
+depends_build-append \
+                    port:pkgconfig
+
+compiler.c_standard 1999
+compiler.cxx_standard \
+                    2017
+
+# These will be overridden by enabled variants
+configure.args-append \
+                    -DBUILD_GUILE_BINDINGS=OFF \
+                    -DBUILD_PYTHON_BINDINGS=OFF \
+                    -DBUILD_STUDIO_APP=OFF
+
+patch {
+    # Distributed binaries should not be built with -march=native
+    reinplace "s|-march=native||g" ${worksrcpath}/CMakeLists.txt
+}
```

Comment:
Converted to a patch. I would prefer not to provide a `native` variant without good evidence that it makes a difference. I do provide one for the `nauty` port (because upstream has it as a config option), but could not actually measure the difference. I considered it for `cliquer`, but after seeing no difference in performance, I gave up on trying to do this in general.